packages/caravan-fees/src/btcTransactionComponents.ts

```diff
@@ -0,0 +1,331 @@
+  }
+
+  /**
+   * Get the amount in BTC
+   * @returns The amount in BTC (as a string)
+   */
+  get amountBTC(): BTC {
+    return satoshisToBTC(this._amountSats.toString());
+  }
+
+  hasAmount(): boolean {
+    return this._amountSats.isGreaterThanOrEqualTo(0);
+  }
+
+  /**
+   * Check if the component is valid
+   * @returns True if valid, false otherwise
+   */
+  abstract isValid(): boolean;
+}
+
+/**
+ * Represents a Bitcoin transaction input template for PSBT creation.
+ * This class contains the minimal required fields and optional fields
+ * necessary for creating a valid PSBT input.
+ *
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki
+ */
+export class BtcTxInputTemplate extends BtcTxComponent {
+  private readonly _txid: string;
+  private readonly _vout: number;
+  private _nonWitnessUtxo?: Buffer;
+  private _witnessUtxo?: {
+    script: Buffer;
+    value: number;
+  };
+  private _sequence?: number;
+
+  /**
+   * @param {Object} params - The parameters for creating a BtcTxInputTemplate
+   * @param {string} params.txid - The transaction ID of the UTXO (reversed, big-endian)
+   * @param {number} params.vout - The output index in the transaction
+   * @param {Satoshis} params.amountSats - The amount in satoshis
+   */
+  constructor(params: { txid: string; vout: number; amountSats?: Satoshis }) {
+    super(params.amountSats || "0");
+    this._txid = params.txid;
+    this._vout = params.vout;
+  }
+  /**
+   * Creates a BtcTxInputTemplate from a UTXO object.
+   */
+  static fromUTXO(utxo: UTXO): BtcTxInputTemplate {
+    const template = new BtcTxInputTemplate({
+      txid: utxo.txid,
+      vout: utxo.vout,
+      amountSats: utxo.value,
+    });
+
+    if (utxo.prevTxHex) {
+      template.setNonWitnessUtxo(Buffer.from(utxo.prevTxHex, "hex"));
+    }
+
+    if (utxo.witnessUtxo) {
+      template.setWitnessUtxo(utxo.witnessUtxo);
+    }
+
+    return template;
+  }
+  /**
+   * The transaction ID of the UTXO (reversed, big-endian).
+   * Required for all PSBT inputs.
+   */
+  get txid(): string {
+    return this._txid;
+  }
+
+  /**
+   * The output index in the transaction.
+   * Required for all PSBT inputs.
+   */
+  get vout(): number {
+    return this._vout;
+  }
+
+  /** Get the sequence number */
+  get sequence(): number | undefined {
+    return this._sequence;
+  }
+
+  /**
+   * Sets the sequence number for the input.
+   * Optional, but useful for RBF signaling.
+   * @param {number} sequence - The sequence number
+   */
+  setSequence(sequence: number): void {
+    this._sequence = sequence;
+  }
+
+  /**
+   * Gets the non-witness UTXO.
+   */
+  get nonWitnessUtxo(): Buffer | undefined {
+    return this._nonWitnessUtxo;
+  }
+
+  /**
+   * Sets the non-witness UTXO.
+   * Required for non-segwit inputs in PSBTs.
+   * @param {Buffer} value - The full transaction containing the UTXO being spent
+   */
+  setNonWitnessUtxo(value: Buffer): void {
+    this._nonWitnessUtxo = value;
```

Comment:
What do you think about doing some validation? Should be able to use the bitcoinjs-lib to just see if a tx correctly loads from the value. Further validation would be confirming that the txid and output index match what's set (if they're set). 

Non-blocking comment, but could be useful.

Reply:
Added a new util function to validate ` setNonWitnessUtxo` and  `setSequence` [here](https://github.com/caravan-bitcoin/caravan/pull/114/commits/272d946f39204f260f8a5f034a78b9f0cdb1336b#diff-18178f7ddc98674c00c89bf5b8c536584ec5aa8f143ed123b4a8b166b2e8a85b)